governance/xc_admin/packages/xc_admin_frontend/utils/parseEvmExecuteCallData.ts

```diff
@@ -0,0 +1,68 @@
+import { decodeFunctionData } from 'viem'
+
+const ABI = [
+  {
+    inputs: [],
+    name: 'acceptOwnership',
+    outputs: [],
+    stateMutability: 'nonpayable',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'acceptAdmin',
+    outputs: [],
+    stateMutability: 'nonpayable',
+    type: 'function',
+  },
+  {
+    inputs: [
+      {
+        internalType: 'address',
+        name: 'newImplementation',
+        type: 'address',
+      },
+    ],
+    name: 'upgradeTo',
+    outputs: [],
+    stateMutability: 'nonpayable',
+    type: 'function',
+  },
+]
+
+/**
+ * Parses the call data for an EVM contract call only if the call data matches one of the entry in ABI.
+ * If there is a match, the method name and inputs are returned. Else, undefined is returned.
+ * @param callData The call data in hex format without leading `0x`.
+ * @returns The parsed call data or undefined.
+ */
+export function parseEvmExecuteCallData(callData: string):
+  | {
+      method: string
+      inputs: [string, string][]
+    }
+  | undefined {
+  try {
+    const { functionName, args } = decodeFunctionData({
+      abi: ABI,
+      data: `0x${callData}`,
+    })
+
+    const abi = ABI.find((abi) => abi.name === functionName)!
+
+    return {
+      method: functionName,
+      inputs: abi.inputs.map((input, i) => {
+        // if inputs are defined, args will be defined
+        let arg = args![i]
+        if (input.type === 'tuple' || Array.isArray(arg)) {
+          return [input.name, JSON.stringify(arg)]
+        }
+
+        return [input.name, arg as string]
+      }),
+    }
+  } catch (e) {
```

Comment:
I am not sure about narrowing down here. In case of any error, we are showing the original call data. 
If we try to handle the error, I am not sure what what will we show on the UI or should we even show anything? 

Also, this would be very rare.